at_commons/lib/src/exception/at_exceptions.dart

```diff
@@ -7,12 +7,32 @@ class AtException implements Exception {
   /// Represents error message that details the cause of the exception
   var message;
 
+  ContextParams? contextParams;
+
   AtException(this.message);
 
   @override
   String toString() {
     return 'Exception: $message';
   }
+
+  String format() {
+    String formattedMessage = message;
+    if (formattedMessage.contains('{key}')) {
+      formattedMessage =
+          formattedMessage.replaceAll('{key}', contextParams!.key);
+    }
+    if (formattedMessage.contains('{currentAtSign}')) {
+      formattedMessage = formattedMessage.replaceAll(
+          '{currentAtSign}', contextParams!.currentAtSign);
+    }
+    if (formattedMessage.contains('{receiverAtSign}') &&
+        contextParams!.receiverAtSign != null) {
+      formattedMessage = formattedMessage.replaceAll(
+          '{receiverAtSign}', contextParams!.receiverAtSign!);
+    }
+    return formattedMessage;
+  }
 }
```

Comment:
The bang operator is asserting that contextParams will never be null when format() is run ... how are you ensuring that contextParams will never be null?